### PR TITLE
 feat: HTTP Status code and handlers(#39522)

### PR DESCRIPTION
**Pull request by ajayrose25**

### CODE REVIEW

**Review by VisaaliSuccessive — CHANGES_REQUESTED** [review]
Have you really run this code,  was this running properly? As I am unable to see any changes in your package.json file.

File: src/Server.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import * as express from 'express';
+import * as bodyParser from 'body-parser';
 import Iconfig from './config/iconfig';
+import { errorHandler, notFoundRoute } from './libs/routes/index';
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: remove index from the path, no need to add index while importing

File: src/Server.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
...
 import { eventNames } from 'cluster';
```
Comment (VisaaliSuccessive) [inline src/Server.ts:5]: why is this used? Please make sure to remove unnecessary importing before pushing

File: src/Server.ts [hunk 3]
```diff
@@ -8,19 +10,29 @@ class Server {
...
     bootstrap() {
+        this.initBodyParser();
         this.setupRoutes();
+
         return this;
     }
     run = () => {
         const { app, config: { port, env } } = this;
-        app.listen(port, (err) => {
+        app.listen(9000, (err) => {
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: don't hard code the port call that from .env file only

File: src/Server.ts [hunk 4]
```diff
@@ -8,19 +10,29 @@ class Server {
...
             if (err) {
                 throw err;
             }
-            console.log('app is running successfully on' , {port}, {env} );
+            console.log('app is running successfully on', { port }, { env });
         });
+        app.use(notFoundRoute);
+        app.use(errorHandler);
```
Comment (VisaaliSuccessive) [inline src/Server.ts:27]: move these to setUpRoutes method

File: src/libs/routes/errorHandler.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const errorHandler = ((err, req, res, next) => {
+    console.log('Error', err);
```
Comment (VisaaliSuccessive) [inline src/libs/routes/errorHandler.ts:2]: Check whether all the headers are sent or not if not then throw error

File: src/libs/routes/errorHandler.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+    res.send({
+        error: 'not found',
+        message: 'error',
+        status: 500,
+        timestamp: new Date()
```
Comment (VisaaliSuccessive) [inline src/libs/routes/errorHandler.ts]: move all these to a variable and pass them into response

File: src/libs/routes/errorHandler.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+const errorHandler = ((err, req, res, next) => {
+    console.log('Error', err);
+    res.send({
+        error: 'not found',
```
Comment (VisaaliSuccessive) [inline src/libs/routes/errorHandler.ts]: add an optional error if we don't get any like:
 error: error || 'undefined'

File: src/libs/routes/errorHandler.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+        message: 'error',
```
Comment (VisaaliSuccessive) [inline src/libs/routes/errorHandler.ts]: add message also if you don't get any error from the error stack also use destructuring

**Review by VisaaliSuccessive — APPROVED** [review]

File: src/Server.ts (hunk 1 above)
Comment (ajayrose25) [reply src/Server.ts]: Done

File: src/Server.ts (hunk 2 above)
Comment (ajayrose25) [reply src/Server.ts:5]: Done

File: src/Server.ts (hunk 3 above)
Comment (ajayrose25) [reply src/Server.ts]: Done

File: src/Server.ts (hunk 4 above)
Comment (ajayrose25) [reply src/Server.ts:27]: Done

File: src/libs/routes/errorHandler.ts (hunk 6 above)
Comment (ajayrose25) [reply src/libs/routes/errorHandler.ts]: Done

File: src/libs/routes/errorHandler.ts (hunk 7 above)
Comment (ajayrose25) [reply src/libs/routes/errorHandler.ts]: Done

File: src/libs/routes/errorHandler.ts (hunk 8 above)
Comment (ajayrose25) [reply src/libs/routes/errorHandler.ts]: Done

File: src/libs/routes/errorHandler.ts (hunk 5 above)
Comment (ajayrose25) [reply src/libs/routes/errorHandler.ts:2]: Done

**Review by VisaaliSuccessive — APPROVED** [review]

**Pull request merged**